docs/Tuple.md

```diff
@@ -0,0 +1,71 @@
+# Tuple
+
+A `Tuple` is a simple data type that represents two types of values at the same
+time. It is typically used to pass two values around without needing to
+introduce some adhoc data type, such as the accumulating state of an operation
+along with the previous result, or coupling an ID along with a set of values.
+
+Note that while other libraries and languages may offer tuples of varying sizes,
+the `Tuple` currently offered here is restricted to holding two values.
+
+## Construction
+
+`Tuple` consists of a single constructor that accepts the two values to store in
+the tuple `Tuple :: a -> b -> Tuple a b`. It is worth highlighting that the two
+values associated with the `Tuple` may be completely different types.
+
+## Interaction
+
+The values of a `Tuple` instance can be accessed by passing the `Tuple` to
+`Tuple.fst` or `Tuple.snd` for the first and second values respectively.
+
+## Reference
+
+### Constructors
+
+#### `Tuple :: (a, b) -> Tuple a b`
+
+### Static functions
+
+#### `Tuple.fst :: Tuple a b -> a`
+
+Returns the first element of a given `Tuple`.
+
+#### `Tuple.snd :: Tuple a b -> b`
+
+Returns the second element of a given `Tuple`.
+
+### Instance methods
+
+#### `tuple.concat :: Semigroup a, Semigroup b => Tuple a b ~> Tuple a b -> Tuple a b`
```

Comment:
I don't mind either way. Probably a good idea to keep with the Haskell-like syntax.